examples/sensors/custom_data_handlers.py

```diff
@@ -0,0 +1,521 @@
+        self.x_data.clear()
+
+    def process(self, data, cur_time):
+        """Process new data point and update plot."""
+        # Auto-detect data structure on first call if not already set up
+        if self.subplot_structure is None:
+            self._setup_plot_structure(data)
+
+        # Convert data to appropriate format
+        if isinstance(data, dict):
+            # Dictionary input - expect dict format
+            if not self.is_dict_data:
+                gs.raise_exception("PyQtGraphPlotter: Expected array data but received dictionary")
+            processed_data = {}
+            for key, values in data.items():
+                if key not in self.subplot_structure:
+                    continue  # Skip keys not in our structure
+                values = np.atleast_1d(tensor_to_array(values))
+                processed_data[key] = values
+        else:
+            # Array input
+            data = np.atleast_1d(tensor_to_array(data))
+            if self.is_dict_data:
+                gs.raise_exception("PyQtGraphPlotter: Expected dictionary data but received array")
+            processed_data = {"main": data}
+
+        # Update time data
+        self.x_data.append(cur_time)
+
+        # Update y data for each subplot
+        for subplot_key, subplot_data in processed_data.items():
+            if subplot_key not in self.y_data:
+                gs.logger.debug(f"PyQtGraphPlotter: Skipping unknown subplot key '{subplot_key}'")
+                continue
+
+            channel_labels = list(self.subplot_structure[subplot_key])
+            if len(subplot_data) != len(channel_labels):
+                gs.logger.warning(
+                    f"PyQtGraphPlotter: Data length ({len(subplot_data)}) doesn't match "
+                    f"number of channels ({len(channel_labels)}) for subplot '{subplot_key}'"
+                )
+                continue
+
+            for i, channel_label in enumerate(channel_labels):
+                if i < len(subplot_data):
+                    self.y_data[subplot_key][channel_label].append(float(subplot_data[i]))
+
+        # Safety check: ensure all channels have same length as x_data
+        for subplot_key in self.y_data:
+            for channel_label in self.y_data[subplot_key]:
+                while len(self.y_data[subplot_key][channel_label]) < len(self.x_data):
+                    # Pad with last value or 0 if empty
+                    last_val = (
+                        self.y_data[subplot_key][channel_label][-1] if self.y_data[subplot_key][channel_label] else 0.0
+                    )
+                    self.y_data[subplot_key][channel_label].append(last_val)
+
+        # Maintain rolling window
+        if len(self.x_data) > self.rolling_window_size:
+            self.x_data.pop(0)
+            for subplot_key in self.y_data:
+                for channel_label in self.y_data[subplot_key]:
+                    if len(self.y_data[subplot_key][channel_label]) > 0:
```

Comment:
`if self.y_data[subplot_key][channel_label]:` is fine.

Moreover, if this condition is very rarely invalid, it would be faster to apply the principle "better ask forgiveness than permission":
```python
try:
    self.y_data[subplot_key][channel_label].pop(0)
except IndexError:
    # Empty sequence nothing to do.
    break
```
Note that I used `break` instead of `pass` because I think if the condition is triggered for one channel, it should be the case for all of them?